### PR TITLE
v1.22.0 Release

### LINKED ISSUE #20: Delivery point selection not working with SSL
Hello,

When Prestashop is set to have SSL enable, but not on all pages, the order confirmation process is secured by SSL.

But the bpostshm module does not consider this. When you have to select a delivery point (either @bpost or @24/7), the module opens a fancybox that calls to non-HTTPS elements. Browser security policy will block all these calls.

So this module is mostly non-working with HTTPS-enabled Prestashops.

### PR DESCRIPTION
**Pull request by Stigmi**
Change-log
* Parcel locker delivery method no longer needs registration
* Settings reorganized by tabs
* New minimal amount condition for options
* New API URL
* SSL issue fixed #20 (thank you @Tito1337)
* Map search issues fixed when no result or only one result
* Display issue fixed on a 1.4 button when validating selection in FR 

*New bpost webservice is online, this release can go live as soon as validated*

### CODE REVIEW

**Comment by Quetzacoalt91** [comment]
Hi,

Why did you create the file *classes/UpgradeService.php* ? You do not need to create a functionnality already existing on PrestaShop ? If it is for PS v1.4, you can have a really simpler version.

See an example here: https://github.com/PrestaShop/ganalytics/blob/aa9430712ed92fb26f220f2d2b1329b146bda041/ganalytics.php#L772-L785

**Comment by Stigmi** [comment]
And the validation process starts :)

 

		
Thomas Percevaut - Stigmi


Fr : +33 617 55 16 14


Be : +32 493 90 02 79

 

De : Thomas N [mailto:notifications@github.com] 
Envoyé : mercredi 5 août 2015 11:03
À : PrestaShop/bpostshm
Cc : Stigmi
Objet : Re: [bpostshm] v1.22.0 Release (#23)

 

Hi,

Why did you create the file classes/UpgradeService.php ? You do not need to create a functionnality already existing on PrestaShop ? If it is for PS v1.4, you can have a really simpler version.

See an example here: https://github.com/PrestaShop/ganalytics/blob/aa9430712ed92fb26f220f2d2b1329b146bda041/ganalytics.php#L772-L785

**Comment by Quetzacoalt91** [comment]
Hi,

You commented a lot of code in this PR, it would be great if you remove the concerned lines.

File: views/templates/hook/back-office-header.tpl [hunk 1]
```diff
@@ -0,0 +1,12 @@
+{*
+* 2014-2015 Stigmi
+*
+* @author Serge <serge@stigmi.eu>
+* @copyright 2014-2015 Stigmi
+* @license http://opensource.org/licenses/osl-3.0.php  Open Software License (OSL 3.0)
+*}
+<link rel="stylesheet" href="{$module_dir|escape}/views/css/admin-bpost.css" type="text/css" />
```
Comment (Quetzacoalt91) [inline views/templates/hook/back-office-header.tpl]: You should use the functions `addCss()` and `addJs()` in order to include these files.

Comment (Stigmi) [inline views/templates/hook/back-office-header.tpl]: Currently, it breaks the tabs of the 1.4 layout if we do this.
settings 1 4

We are currently working on the back office for version 1.3, so we'll look into calling these files more selectively. One of the issues we came across is that addJS and addCSS are controller functions >> not available in every hook and part of context.

Comment (Stigmi) [inline views/templates/hook/back-office-header.tpl]: Regarding comments, 1.22 is an intermediate version in which we added some features / options that are yet not visible (commented).

We'll uncomment / clean remaining old comments for 1.3.
Right now we need them to stay in the code.
Thx

**Comment by Stigmi** [comment]
Regarding classes/UpgradeService.php, it is indeed for 1.4

We tried your suggestion and we ended up with duplicate db columns and strange issues in 1.4.
We may have taken the long way, but it's also the safe way.

**Pull request merged**